### PR TITLE
Enable to overwrite the Kubernetes version for CSI migration on Shoot level

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area storage
/kind enhancement
/platform aws

**What this PR does / why we need it**:
This PR adds support for overwriting the Kubernetes version for which CSI migration shall be performed on `Shoot` level.

⚠️ This feature is to be considered alpha/experimental and usage is on your own risk.

**Special notes for your reviewer**:
/squash
✅ Depends on gardener/gardener#4202

**Release note**:

```other operator
Support for overwriting the CSI migration version was added.
```

### CODE REVIEW

**Review by timuthy — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/coreos/go-systemd/v22 v22.1.0
 	github.com/frankban/quicktest v1.9.0 // indirect
 	github.com/gardener/etcd-druid v0.5.0
-	github.com/gardener/gardener v1.24.1-0.20210608063816-580010846480
+	github.com/gardener/gardener v1.24.1-0.20210611132013-a6aa5c0904a9
```
Comment (timebertt) [inline go.mod:13]: I assume, you will cherry-pick this and vendor `v1.23.2` in the cherry-pick PR, right?

**Review by timebertt — APPROVED** [review]
/lgtm

File: go.mod (hunk 1 above)
Comment (rfranzke) [reply go.mod:13]: Yes, https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-release-v1.23/jobs/release-v1.23-release-job/builds/4 is running and I'll open a PR for `release-v1.25` once `gardener/gardener@v1.23.2` is released successfully :)

Comment (timebertt) [reply go.mod:13]: Great 🚀

**Pull request merged**